tanner/tests/test_api.py

```diff
@@ -2,348 +2,293 @@
 import asyncio
 import aioredis
 import itertools
-
+import sqlalchemy
 from unittest import mock
 from tanner.api.api import Api
-from tanner import redis_client
+from tanner import postgres_client
+from tanner.dbutils import DBUtils
 from tanner.utils.asyncmock import AsyncMock
 
+SESSION_DATA = {
+    "sess_uuid": "ba800b95-28dd-4a78-b279-940781ce9513",
+    "peer_ip": "196.207.97.20",
+    "peer_port": 36864,
+    "location": {
+        "country": "India",
+        "country_code": "IN",
+        "city": "Delhi",
+        "zip_code": "110092",
+    },
+    "user_agent": "Mozilla/5.0",
+    "snare_uuid": "9f7d7dd3-ac6b-468b-8cee-ce3e352eff6e",
+    "start_time": 1589192688.001088,
+    "end_time": 1589192688.7164452,
+    "requests_in_second": 6.989514155656564,
+    "approx_time_between_requests": 0.14308667182922363,
+    "accepted_paths": 5,
+    "errors": 0,
+    "hidden_links": 0,
+    "attack_types": ["index", "lfi", "cmd_exec"],
+    "attack_count": {"index": 5},
+    "paths": [
+        {
+            "path": "/sites/default/files",
+            "timestamp": 1589192688.0010145,
+            "response_status": 200,
+            "attack_type": "index",
+        },
+        {
+            "path": "/sites/default/../files",
+            "timestamp": 1589192658.0010125,
+            "response_status": 200,
+            "attack_type": "lfi",
+        },
+        {
+            "path": "/sites/default/;ls",
+            "timestamp": 1589192618.0010145,
+            "response_status": 200,
+            "attack_type": "cmd_exec",
+        },
+    ],
+    "cookies": {"sess_uuid": "d96bfa6d-e7a4-4344-99ab-c39cc448208f"},
+    "referer": "/",
+    "possible_owners": {"user": 1},
+}
+
 
 class TestApi(unittest.TestCase):
     def setUp(self):
         self.loop = asyncio.new_event_loop()
-        self.redis_client = None
-        self.snare_uuid = "9a631aee-2b52-4108-9831-b495ac8afa80"
-        self.uuid = "da1811cd19d748058bc02ee5bf9029d4"
+        self.pg_client = None
         self.returned_content = None
         self.expected_content = None
         self.conn = None
         self.key = None
 
-        async def connect():
-            self.redis_client = await redis_client.RedisClient.get_redis_client()
+        async def create_db():
+            with sqlalchemy.create_engine(
+                "postgresql://postgres@/postgres", isolation_level="AUTOCOMMIT"
+            ).connect() as connection:
+                connection.execute("CREATE DATABASE tanner_test_db")
+                connection.close()
 
+        async def connect():
+            self.postgres = postgres_client.PostgresClient()
```

Comment:
I am just wondering, where this variable is used?

Reply:
After this line, we are defining all the variables required to connect to Postgres.
```
        async def connect():
            self.postgres = postgres_client.PostgresClient()
            self.postgres.host = "localhost"
            self.postgres.post = 5432
            self.postgres.db_name = "tanner_test_db"
            self.postgres.user = "postgres"
            self.postgres.password = "postgres"
            self.postgres.maxsize = 80

```

Reply:
🤦 my bad :)